tests/test_presolve.py

```diff
@@ -40,6 +41,36 @@ class TestNormalization(unittest.TestCase):
     # Developer note: most of the testing for normalization is done at the C++
     # level. These tests are mostly testing the Cython/Python interface.
 
+    def test_concurrency(self):
+        # obviously this is difficult to test fully, but we can at least run some
+        # smoke tests to try to detect deadlocks and corrupted models
+
+        cqm = dimod.ConstrainedQuadraticModel()
+        cqm.add_variables("BINARY", 1000)
+        for _ in range(1000):
+            # these will be flipped, so if we're modifying the same model in
+            # several threads at the same time we should see some issues
+            cqm.add_constraint(((v, 1) for v in range(1000)), '>=', 1)
+
+        presolver = Presolver(cqm)
+
+        num_threads = 4
```

Comment:
Perhaps set to `cpu_count - 1`, or at least to `min(4, cpu_count)`. By [`cpu_count`](https://docs.python.org/3/library/os.html#os.cpu_count) I mean `len(os.sched_getaffinity(0))`.